Should `jl_get_ptls_states_wrapper` be extern instead of static?

Currently get this compiler warning when compiling the master branch:
```
.../julia/src/threading.c:187:32: warning: ‘jl_get_ptls_states_wrapper’ defined but not used [-Wunused-function]
  187 | static JL_CONST_FUNC jl_ptls_t jl_get_ptls_states_wrapper(void) JL_GLOBALLY_ROOTED JL_NOTSAFEPOINT
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
```
There are no references to `jl_get_ptls_states_wrapper` anywhere I could find, and anyway, it is static. So it could be removed. Except that one comment suggests that it is meant to be accessible by code linking against libjulia. Except it isn't, as far as I can tell. An oversight?

Perhaps I am wrong and this *is* accessible, via some black magic. Then I still think there is an issue here: the code really should have a comment explaining this.

--- a/src/threading.c
+++ b/src/threading.c
@@ -142,12 +142,10 @@ jl_get_ptls_states_func jl_get_ptls_states_getter(void)
 // We use the faster static version in the main executable to replace
 // the slower version in the shared object. The code in different libraries
 // or executables, however, have to agree on which version to use.
-// The general solution is to add one more indirection in the C entry point
-// (see `jl_get_ptls_states_wrapper`).
+// The general solution is to add one more indirection in the C entry point.
 //
 // When `ifunc` is available, we can use it to trick the linker to use the
 // real address (`jl_get_ptls_states_static`) directly as the symbol address.
-// (see `jl_get_ptls_states_resolve`).
 //
 // However, since the detection of the static version in `ifunc`
 // is not guaranteed to be reliable, we still need to fallback to the wrapper
@@ -182,13 +180,6 @@ static jl_ptls_t jl_get_ptls_states_init(void)
 #  endif
     jl_tls_states_cb = cb;
     return cb();
-}
-
-static JL_CONST_FUNC jl_ptls_t jl_get_ptls_states_wrapper(void) JL_GLOBALLY_ROOTED JL_NOTSAFEPOINT
-{
-#ifndef __clang_analyzer__
-    return (*jl_tls_states_cb)();
-#endif
 }
 
 JL_DLLEXPORT void jl_set_ptls_states_getter(jl_get_ptls_states_func f)

